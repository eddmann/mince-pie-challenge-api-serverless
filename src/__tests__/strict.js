import handler from '../handlers/strict';

it('fails to respond when access token not found', async () => {
  const services = { getUserIdFromToken: () => Promise.resolve() };

  const response = parseResponse(await handler(services)({ headers: { Authorization: 'TOKEN' } }, {}));

  expect(response.statusCode).toBe(401);
  expect(response.body.title).toBe('Unauthorized');
});

it('successfully responds with a user id when access token found', async () => {
  const services = { getUserIdFromToken: () => Promise.resolve('USER_ID') };

  const response = parseResponse(await handler(services)({ headers: { Authorization: 'TOKEN' } }, {}));

  expect(response.statusCode).toBe(200);
  expect(response.body.userId).toBe('USER_ID');
});
