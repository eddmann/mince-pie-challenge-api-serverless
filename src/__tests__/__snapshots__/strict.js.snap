// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fails to respond when access token not found 1`] = `
Object {
  "body": "{\\"title\\":\\"Unauthorized\\",\\"detail\\":\\"Service requires an authenticated user\\"}",
  "headers": Object {
    "Access-Control-Allow-Credentials": "*",
    "Access-Control-Allow-Origin": "*",
    "Content-Type": "application/problem+json",
  },
  "statusCode": 401,
}
`;

exports[`successfully responds with a user id when access token found 1`] = `
Object {
  "body": "{\\"userId\\":\\"USER_ID\\"}",
  "statusCode": 200,
}
`;
